packages/inquirer/src/ui/prompt.mts

```diff
@@ -389,10 +409,7 @@ export default class PromptsRunner<A extends Answers> {
   };
 
   private shouldRun = async (question: AnyQuestion<A>): Promise<boolean> => {
-    if (
-      question.askAnswered !== true &&
-      _.get(this.answers, question.name) !== undefined
-    ) {
+    if (question.askAnswered !== true && this.answers[question.name] !== undefined) {
```

Comment:
This will break where `name` contains a sub-path like `name: 'foo.bar'` (which is why `_.get` was used)

Reply:
It's using a proxy.
Tests are in place for sub-paths.

Reply:
Oh, I missed that. I think it's fair to do internally, but I'm unclear why we want to expose the capacity to provide a proxy as argument?

Reply:
I want to provide a custom store.
It's possible to use a proxy or implement an api.

This is actually required to simplify supporting untouched inquirer api.
So Observables are supported by loading the question answer on demand.
Currently the prefilled answers are built from questions array.

Reply:
This is a feature request, I can drop it.

Reply:
I'm not against it if there's a use-case. Just want to make sure we don't add unnecessary complexity we then need to support in "perpetuity". Or open doors for people to unnecessarily overcomplexify their implementations.